File: TodoAppJava/app/src/main/java/com/cwtstudio/todoapp/FileHelper.java
package com.cwtstudio.todoapp;

import android.content.Context;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.util.ArrayList;

/**
 * CodeWithTamim
 *
 * @developer Tamim Hossain
 * @mail tamimh.dev@gmail.com
 */
public class FileHelper {
    private static final String FILE_NAME = "listInfo.dat";

    public static void writeFile(ArrayList<String> itemList, Context context) {
        try {
            FileOutputStream fos = context.openFileOutput(FILE_NAME, Context.MODE_PRIVATE);
            ObjectOutputStream objOutPutStream = new ObjectOutputStream(fos);
            objOutPutStream.writeObject(itemList);
            objOutPutStream.close();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public static ArrayList<String> readFile(Context context) {
        try {

            FileInputStream fis = context.openFileInput(FILE_NAME);
            ObjectInputStream inputStream = new ObjectInputStream(fis);
            return (ArrayList<String>) inputStream.readObject();
        } catch (Exception e) {
            return new ArrayList<>();
        }
    }


}

File: TodoAppKotlin/app/src/main/java/com/cwtstudio/todokotlin/FileHelper.kt
package com.cwtstudio.todokotlin

import android.content.Context
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

/**
 * CodeWithTamim
 *
 * @developer Tamim Hossain
 * @mail tamimh.dev@gmail.com
 */
object FileHelper {
    private const val FILE_NAME: String = "listInfo.dat"

    fun writeItem(itemList: ArrayList<String>, context: Context) {
        try {
            val outputStream: FileOutputStream =
                context.openFileOutput(FILE_NAME, Context.MODE_PRIVATE)
            val objectOutPutStream: ObjectOutputStream = ObjectOutputStream(outputStream)
            objectOutPutStream.writeObject(itemList)
            objectOutPutStream.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }


    fun readItem(context: Context): ArrayList<String> {
        try {
            val inputStream: FileInputStream = context.openFileInput(FILE_NAME)
            val objectInputStream = ObjectInputStream(inputStream)
            return objectInputStream.readObject() as ArrayList<String>

        } catch (e: Exception) {
            return ArrayList<String>()
            e.printStackTrace()
        }

    }


}